app/models/story.rb

```diff
@@ -30,6 +32,7 @@ class Story < ActiveRecord::Base
   has_many :notifications, as: :source, dependent: :destroy
 
   validates :user, :story_type, presence: true
+  validates :group, presence: true, if: Proc.new {|s| s.group_id.present? }
```

Comment:
If the ID is present then it validates the relationship, as in it's an actual valid object.